src/Control/Monad/Static.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE StaticPointers #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Control.Monad.Static where
+
+import Control.Applicative.Static
+import Control.Distributed.Closure
+import Data.Functor.Static
+import Data.Typeable (Typeable)
+
+-- | Instances of 'StaticBind' should satisfy the following laws (writing
+-- 'staticMap', 'staticApply', 'staticBind' as infix @('<$>')@, @('<*>')@, @(>>=)@,
+-- respectively):
+--
+-- @
+-- (m >>= f) >>= g = m >>= static (.) ``cap`` (staticFlippedBind g) `cap` f
```

Comment:
Interesting. `staticFlipBind` is the more general version of `staticBind` that I've mentioned before. You can't actually define it for an arbitrary `StaticBind m`, as the constraint is not known to be `Static`. 

(the same set of remarks holds for `staticJoin` below)